assets/tests/testHelpers/socketHelpers.ts

```diff
@@ -1,23 +1,21 @@
-import { Socket } from "phoenix"
+import { jest } from "@jest/globals"
+import { Socket, PushStatus, Channel } from "phoenix"
 
 export const makeMockSocket = (): Socket & { channel: jest.Mock } =>
   ({
     channel: jest.fn(),
   } as Socket & { channel: jest.Mock })
 
 export const makeMockChannel = (
-  expectedReceiveMessage?:
-    | "ok"
-    | "error"
-    | "timeout"
-    | (() => "ok" | "error" | "timeout"),
+  expectedReceiveMessage?: PushStatus | (() => PushStatus),
   expectedReceiveData?: (() => any) | any
 ) => {
   const result = {
     join: jest.fn(),
     leave: jest.fn(),
-    on: jest.fn(),
-    receive: jest.fn(),
+    on: jest.fn<Channel["on"]>(),
+    receive:
+      jest.fn<(message: PushStatus, handler: (data?: any) => void) => void>(),
```

Comment:
issue(non-blocking): it seems like the API we have here isn't quite right? From what I can tell, `receive` doesn't exist on the `Channel` type, but is instead on the `Push` type, which is returned by the `join`, `leave`, and `push` methods. I'm thinking we should open a tech debt ticket to fix this? 

I'd like to know what you think and if this is something we should revisit in the future.

Reply:
Ticket [here](https://app.asana.com/0/1152340551558956/1205307050135117/f).